### PR TITLE
refactor initRegistryAndConfig function

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Resolve this TODO task together with some refactoring works.
https://github.com/kubernetes/kubernetes/blob/9d41c13348f4ef0c7aaa1ae5362a6a78d574b179/test/integration/scheduler/framework_test.go#L2282

#### Which issue(s) this PR fixes:

Part of #https://github.com/kubernetes/kubernetes/issues/92229 as pre-work

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/scheduler/framework_test.go [hunk 1]
```diff
@@ -2289,15 +2168,41 @@ func initRegistryAndConfig(t *testing.T, pp ...*PermitPlugin) (frameworkruntime.
...
-		registry.Register(p.Name(), newPermitPlugin(p))
-		versionedCfg.Profiles[0].Plugins.Permit.Enabled = append(versionedCfg.Profiles[0].Plugins.Permit.Enabled, v1beta3.Plugin{Name: p.Name()})
+		// register plugin
+		switch pl := p.(type) {
+		case *PermitPlugin:
+			registry.Register(p.Name(), newPermitPlugin(pl))
+		default:
+			registry.Register(p.Name(), newPlugin(p))
+		}
+
```
Comment (denkensk) [inline test/integration/scheduler/framework_test.go:2183]: Make code cleaner?
```go
plugins := &v1beta3.Plugins{}
plugin := v1beta3.Plugin{Name: p.Name()}
switch p.(type) {
    .......
    plugins.PreFilter.Enabled = append(plugins.PreFilter.Enabled, plugin)
}

versionedCfg.Profiles[0].Plugins = plugins
```

**Review by denkensk — COMMENTED** [review]
Thanks @kerthcet  
some comments

File: test/integration/scheduler/framework_test.go (hunk 1 above)
Comment (kerthcet) [reply test/integration/scheduler/framework_test.go:2183]: hmmm, I should review more carefully. Thanks Alex.

**Comment by kerthcet** [comment]
squashed commits.

**Comment by denkensk** [comment]
/lgtm

Thanks @kerthcet 
ping  @alculquicondor for apporve

File: test/integration/scheduler/framework_test.go [hunk 2]
```diff
@@ -2279,26 +2159,53 @@ func initTestSchedulerForFrameworkTest(t *testing.T, testCtx *testutils.TestCont
...
 		return frameworkruntime.Registry{}, schedulerconfig.KubeSchedulerProfile{}
 	}
 
+	registry := frameworkruntime.Registry{}
+	plugins := &v1beta3.Plugins{}
+
+	for _, p := range pp {
+		// register plugin
+		switch pl := p.(type) {
+		case *PermitPlugin:
```
Comment (alculquicondor) [inline test/integration/scheduler/framework_test.go:2173]: I don't quite like this switch. What about accepting the builder instead?

```golang
func initRegistryAndConfig(t *testing.T, plBuilder ...framework.Builder)
```

Comment (kerthcet) [reply test/integration/scheduler/framework_test.go]: I moved this logic to `newPlugin` and removed other plugin initialization functions, any advices?

**Comment by alculquicondor** [comment]
Yeah, this looks better

/approve

Please squash

**Pull request merged**